src/lib/index.js

```diff
@@ -1008,6 +1008,60 @@ lib.templateString = function(string, obj) {
     });
 };
 
+/*
+ * Substitute values from an object into a string and optionally formats them using d3-format,
+ * or fallback to associated labels.
+ *
+ * Examples:
+ *  Lib.templateString('name: %{trace}', {trace: 'asdf'}) --> 'name: asdf'
+ *  Lib.templateString('name: %{trace[0].name}', {trace: [{name: 'asdf'}]}) --> 'name: asdf'
+ *  Lib.templateString('price: %{y:$.2f}', {y: 1}) --> 'price: $1.00'
+ *
+ * @param {string}  input string containing %{...:...} template strings
+ * @param {obj}     data object containing fallback text when no formatting is specified, ex.: {yLabel: 'formattedYValue'}
+ * @param {obj}     data objects containing substitution values
+ *
+ * @return {string} templated string
+ */
+
```

Comment:
another :cow2: (minor coding stylez) valid jsDocs start with `/**` nor `/*` like plain multi-line comments and usually there's no newline between the trainling `*/` and the start of the `function` .

Reply:
Fixed in 8f440b0